### PR TITLE
Explicit Variable Declarations

### PR DESCRIPTION
**Pull request by seadoggie01**
Removed all instances of `Dim` as it is depreciated.
Updated all global variables to be declared once at the top of the file.
Added local scope to all local variables
Removed a few unused variables

(Also, there were two syntax errors... a missing underscore from `_MakeCommand` and a missing closing parenthesis)

### CODE REVIEW

File: UniExtract.au3 [hunk 1]
```diff
@@ -692,21 +694,23 @@ Func ReadPrefs()
...
-	Global $aDefDirs[] = [$userDefDir, $defdir]
-	Global $fileScanLogFile = $logdir & "filescan.txt"
-	Global Const $sPasswordFile = $settingsdir & "\passwords.txt"
+	$prefs = $settingsdir & "\UniExtract.ini"
+	$batchQueue = $settingsdir & "\batch.queue"
+	$logdir = $settingsdir & "\log\"
+	$userDefDir = $settingsdir & "\def\"
+	ReDim $aDefDirs[2]
+	$aDefDirs[0] = $userDefDir
+	$aDefDirs[1] = $defdir
```
Comment (seadoggie01) [inline UniExtract.au3]: This has the same affect as `Global $aDefDirs[] = [$userDefDir, $defDir]`, it just doesn't re-declare the global variable.

File: UniExtract.au3 [hunk 2]
```diff
@@ -2270,7 +2281,7 @@ Func extract($arctype, $arcdisp = 0, $additionalParameters = "", $returnSuccess
 
 		Case $TYPE_CAB
 			If StringInStr($sFileType, 'Type 1', 0) Then
-				RunWait(Warn_Execute(Quote($file & '" /q /x:"' & $outdir), $outdir)
```
Comment (seadoggie01) [inline UniExtract.au3]: This line was missing a closing parenthesis

File: UniExtract.au3 [hunk 3]
```diff
@@ -2504,7 +2516,7 @@ Func extract($arctype, $arcdisp = 0, $additionalParameters = "", $returnSuccess
 
 					; If successful, extract contents of InstallShield cabs file-by-file
 					If $return > 0 Then
-						RunWait(MakeCommand($is6cab & ' x "' & $file & '"', True), $outdir, @SW_MINIMIZE)
```
Comment (seadoggie01) [inline UniExtract.au3]: And here's the `MakeCommand` that's missing an underscore

File: UniExtract.au3 [hunk 4]
```diff
@@ -3533,9 +3548,9 @@ Func HasPlugin($sPlugin, $returnFail = False)
 	_GUICtrlCreatePic($sLogoFile, 8, 20, 49, 49)
 	GUISetState(@SW_SHOW)
 
-	While 1
-		$nMsg = GUIGetMsg()
-		Switch $nMsg
```
Comment (seadoggie01) [inline UniExtract.au3]: This was changed to `While True` for readability.
`$nMsg` was removed as it used once.
(There are multiple places where I changed these)

**Comment by Bioruebe** [comment]
Sorry for ignoring your pull request for so long, I wanted to release a new update before doing bigger code changes - and that took longer than expected...

Thanks a lot for the improvements you made, really good work! I merged most of them, except a few minor things like the global consts. I think it's more readable to keep them as implicit globals.

**Pull request merged**